### PR TITLE
hv: fix size issue in mptable guest copy - ACRN partition mode

### PR DESCRIPTION
**Pull request by sainath14**
1)Fix to take into account the size of proc entries in MPtable.
2)Change APIC ids to virtual APIC IDs

Signed-off-by: Sainath Grandhi <sainath.grandhi@intel.com>

### CODE REVIEW

File: hypervisor/arch/x86/guest/mptable.c [hunk 1]
```diff
@@ -292,12 +292,16 @@ int mptable_build(struct vm *vm)
...
 	struct mpfps            *mpfp;
+	size_t			mptable_length, table_length;
 
 	startaddr = (char *)GPA2HVA(vm, MPTABLE_BASE);
+
+	table_length = vm->vm_desc->mptable->mpch.base_table_length;
+	mptable_length = sizeof(struct mpfps) + table_length;
 	/* Copy mptable info into guest memory */
-	(void)memcpy_s((void *)startaddr, sizeof(struct mptable_info),
+	(void)memcpy_s((void *)startaddr, MPTABLE_MAX_LENGTH,
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/mptable.c]: Why MPTABLE_MAX_LENGTH is (65536 + 16)U ?
And I am not sure if (65536 + 16)U is an valid expression.

**Comment by sainath14** [comment]
the length field in MP config header is 16 bit long. So that should be the reason the max length is set to 65536 bytes. And 16 bytes is the size of MP floating pointer structure. So together it is 65536+16. We can add logic to determine the length and reserve only the exact length in e820. Right now we are reserving 65536+16 in e820.

File: hypervisor/arch/x86/guest/mptable.c (hunk 1 above)
Comment (anthonyzxu) [reply hypervisor/arch/x86/guest/mptable.c]: sainath fixed it.

**Comment by wenlingz** [comment]
Verified +1

**Pull request merged**